### PR TITLE
Split out file loading from astc_main

### PR DESCRIPTION
**Pull request by solidpixel**
Pre-work for migrating to the new API, making astc_main less monolithic.

### CODE REVIEW

File: Source/astcenccli_toplevel.cpp [hunk 1]
```diff
@@ -399,6 +400,165 @@ static void test_inappropriate_cpu_extensions()
...
+}
+
+/**
+ * @brief Load a non-astc image file from memory.
+ *
+ * @param filename           The file to load, or a pattern for array loads.
+ * @param dim_z              The number of slices to load.
+ * @param padding            The number of texels of padding.
+ * @param y_flip             Should this image be Y flipped?
+ * @param linearize_srgb     Should this image be Y flipped?
```
Comment (ghufranjavaid) [inline Source/astcenccli_toplevel.cpp:442]: Should this image converted from linear to sRGB?

File: Source/astcenccli_toplevel.cpp [hunk 2]
```diff
@@ -399,6 +400,165 @@ static void test_inappropriate_cpu_extensions()
...
+			fill_image_padding_area(image);
+		}
+
+		for (auto &i : slices)
+		{
+			free_image(i);
+		}
+	}
+
+	img_num_components = load_result & 7;
```
Comment (ghufranjavaid) [inline Source/astcenccli_toplevel.cpp:556]: Why are we multiplying it with 7? would be a good idea to have it as a const with a comment

Comment (solidpixel) [reply Source/astcenccli_toplevel.cpp]: The original code returns a packed int (bit[2..0] = num components, bit[3]  = 16 bit if set). Seems like needless optimization, so made that long-hand in the next commit.

File: Source/astcenccli_image_load_store.cpp [hunk 3]
```diff
@@ -1468,7 +1456,8 @@ astc_codec_image* load_dds_uncompressed_image(
 
 	delete[] buf;
 	fill_image_padding_area(astc_img);
-	*result = components + (bitness == 16 ? 0x80 : 0);
+	is_hdr = bitness;
```
Comment (solidpixel) [inline Source/astcenccli_image_load_store.cpp:1459]: Bug. Needs bitness == 16.

**Pull request merged**